Allow passing read name to mappy

rob-p opened the pull request:
This adds the (optional) ability to pass the read
name to the mappy `map` method.  Without the
read name, the call to `map` can sometimes give
different output than the command line version
of `minimap2` because of the way minimap uses
the hash of the read name to break ties in ordering hits.  
This can affect which / if certain supplementary alignments 
are generated, and even which / if non-primary alignments 
are generated.

lh3 commented:
Thanks a lot for this. Could you merge `mm_map_aux_with_name()` into `mm_map_aux()`? This is an internal function and API changes are ok.

rob-p commented:
Ok, I've changed the implementation now so that `mm_map_aux` simply takes the name parameter (which can be null) and that gets propagated to the underlying minimap2 calls. I've removed `mm_map_aux_with_name`.

lh3 commented:
Thank you!

rob-p commented:
My pleasure! Thanks as always for the great tools ;).
